index.js

```diff
@@ -351,31 +357,21 @@ function delay(ms) {
  * Given an error from Got, see if Shopify told us how long to wait before
  * retrying.
  *
- * @param {Object} error Error object from Got call
+ * @param {Object} retryObject Got's input for the retry logic
  * @return {(Number|null)} The duration in ms, or `null`
  * @private
  **/
-function maybeRetryMS(error) {
-  // For simplicity, retry network connectivity issues after a hardcoded 1s.
-  if (retryableErrorCodes.has(error.code)) {
-    return 1000;
-  }
-
+function maybeRetryMS(retryObject) {
+  const { error, computedValue } = retryObject;
   const response = error.response;
 
-  if (response.headers && response.headers['retry-after']) {
-    return response.headers['retry-after'] * 1000 || null;
-  }
-
-  if (retryableStatusCodes.has(response.statusCode)) {
-    // Arbitrary 2 seconds, in case we get a 429 without a `Retry-After`
-    // response header, or 4xx/5xx series error that matches the Got retry
-    // defaults.
-    return 2 * 1000;
```

Comment:
We might return `null` instead and use the default `computedValue`. Another option is to still return 2 sec and remove the unreachable code (the last `return null` and the fallback `retryObject.computedValue` in `calculateDelay()`).